### PR TITLE
Removed edge customizations in favor of adding more CSS logic to styleEdge component

### PR DESCRIPTION
**Pull request by leandroberetta**
Resolves https://github.com/kiali/openshift-servicemesh-plugin/issues/223

This PR aims to remove the customizations done to the edge component.

This PR uses the DefaultEdge from PF and it's modified by CSS in the StyleEdge component managed by Kiali (can be renamed to KialiEdge really).

### CODE REVIEW

**Review by jshaughn — COMMENTED** [review]
This is promising, it seems like you may be able to eliminate the need for us to copy DefaultEdge.  But there are some things that have changed or have been lost.

- Edge width has changed.  I'm actually OK thinning the edges a bit, but I don't think it's intentional.  I think this is because if `pathStyle` is supplied we are not applying the entire style, only color.  I think we should probably be honoring the entire style (note - the terminator needs only the color).

- We lost the unhighlighting support
  - Before:
image

  - After:
image

- We lost the `add showTag prop and show scaled tag on hover (when showTag is false)`  customization.

**Comment by leandroberetta** [comment]
> This is promising, it seems like you may be able to eliminate the need for us to copy DefaultEdge. But there are some things that have changed or have been lost.
> 
> * Edge width has changed.  I'm actually OK thinning the edges a bit, but I don't think it's intentional.  I think this is because if `pathStyle` is supplied we are not applying the entire style, only color.  I think we should probably be honoring the entire style (note - the terminator needs only the color).
> * We lost the unhighlighting support
>   
>   * Before:
>     image
>   * After:
>     image
> * We lost the `add showTag prop and show scaled tag on hover (when showTag is false)`  customization.

The edge width is intentional, I'm using 2 I guess now, I think it's cleaner. 

The rest, yes, is missing, I forgot why this was a draft :). I will add it back.

**Review by jshaughn — CHANGES_REQUESTED** [review]
The pathStyle issue is fixed, and I like the thinner edges.  The unhighlighting is better but there is still a problem with unhighlighting the tags:

image

**Comment by leandroberetta** [comment]
> The pathStyle issue is fixed, and I like the thinner edges. The unhighlighting is better but there is still a problem with unhighlighting the tags:
> 
> image

Good catch, now is fixed!

image

**Review by jshaughn — CHANGES_REQUESTED** [review]
Edge tab unhighlighting is fixed, thanks!  But, two more things.  Hovering over an edge loses the edge coloring, and also does not trigger the unhighlighting:

image

**Comment by leandroberetta** [comment]
> Edge tab unhighlighting is fixed, thanks! But, two more things. Hovering over an edge loses the edge coloring, and also does not trigger the unhighlighting:
> 
> image

Interesting, it seems to be a problem when hovering, this should be related to the override of css properties. Thanks.

File: frontend/src/pages/GraphPF/styles/styleEdge.tsx
Comment (ferhoyos) [inline frontend/src/pages/GraphPF/styles/styleEdge.tsx:30]: Please use kialiStyle wrapper instead of style. It adds a prefix to CSS class name to fulfill Openshift plugin CSS requirement for OSSMC

File: frontend/src/pages/GraphPF/styles/styleEdge.tsx
Comment (ferhoyos) [inline frontend/src/pages/GraphPF/styles/styleEdge.tsx:127]: Instead of `join(' ')` I would use `classes` method of `typestyle` library, a utility to join CSS classes conditionally
```
<DefaultEdge className={classes(...cssClasses)} element={element} {...rest} {...passedData} />;
```

**Review by ferhoyos — CHANGES_REQUESTED** [review]

**Comment by leandroberetta** [comment]
> > Edge tab unhighlighting is fixed, thanks! But, two more things. Hovering over an edge loses the edge coloring, and also does not trigger the unhighlighting:
> > image
> 
> Interesting, it seems to be a problem when hovering, this should be related to the override of css properties. Thanks.

@jshaughn I found the issue, there is a missing part due to the removal of the custom edge code, but sadly, I don't see yet how I can do it without touching the edge. At some point, in the edge, when hover is true, we are calling a Kiali function:

https://github.com/kiali/kiali/blob/master/frontend/src/pages/GraphPF/components/edge.tsx#L105

**Comment by jshaughn** [comment]
> but sadly, I don't see yet how I can do it without touching the edge. At some point, in the edge, when hover is true, we are calling a Kiali function:

Right, this is what I meant in the file comment:

`//   add showTag prop and show scaled tag on hover (when showTag is false)`

We need to decide what to do for this PR.  We can:
1. Eliminate our custom `Edge.tsx` and regress with respect to this tag feature.
2. Re-introduce a more minimal `Edge.tsx`, that maintains the feature while we work with PFT for a supported alternative.  

My preference would probably be option 2 because in addition to `Edge.tsx` we still have out customized versions for Node and Groups, so it's not the end of our customization.  I think we should go through the same exercise for Nodes and Groups, seeing what we can move to the `Style` classes, and work with PFT for anything remaining.

**Comment by leandroberetta** [comment]
@jshaughn I finally could make the unhighlighting part:

image

I also fixed the edge style when hovering.

@ferhoyos you're feedback is addressed too. Thanks!.

File: frontend/src/pages/GraphPF/styles/styleEdge.tsx
Comment (ferhoyos) [inline frontend/src/pages/GraphPF/styles/styleEdge.tsx:41]: According to `typestyle` documentation (https://typestyle.github.io/#/core), it is best practice to add symbol `$` to overwrite third party nested CSS classes:

```
const container = style({
  $nest: {
    '& .third-party-class': {
      color: 'red'
    }
  }
});
```
Most of the times it works fine without the `&` symbol, but not in all cases (if the third party CSS class has more specifity than yours). More information here: https://github.com/kiali/kiali/pull/6562

**Review by ferhoyos — CHANGES_REQUESTED** [review]

File: frontend/src/pages/GraphPF/styles/styleEdge.tsx
Comment (ferhoyos) [inline frontend/src/pages/GraphPF/styles/styleEdge.tsx:50]: Each selector is independent, so you have to add '&' for each selector:
`'& .pf-topology__edge__link, & .pf-topology-connector-arrow': data.pathStyle`

**Review by ferhoyos — DISMISSED** [review]
LGTM from the CSS point of view, you have learnt a lot in the process and done a great job @leandroberetta ;-). Anyway I would wait for @jshaughn review from topology features point of view.

File: frontend/src/pages/GraphPF/styles/styleEdge.tsx
Comment (jshaughn) [inline frontend/src/pages/GraphPF/styles/styleEdge.tsx:15]: This is needed to support the lock icons on edge tags.

File: frontend/src/pages/GraphPF/styles/styleEdge.tsx
Comment (jshaughn) [inline frontend/src/pages/GraphPF/styles/styleEdge.tsx:141]:
```suggestion
      <DefaultEdge className={classes(...cssClasses)} element={element} tagClass={tagClass} {...rest} {...passedData} />
```
This is needed to support the lock icons on edge tags.

**Review by jshaughn — CHANGES_REQUESTED** [review]
I think all of the prior comments are resolved!  And we have decided to temporarily remove the tag hide/hover customization and contribute that feature to PFT itself.  So that feature is expected to not be there.

There is a regression in showing lock icons on edge labels, this can be fixed by accepting the recommended changes in my comments.

Also, I'm seeing maybe an issue in trace overlay, but I don't think it has to do with this PR, it may be nothing, but it seems sometimes the overlay stops before the selected node.  Anyway, not a blocker for this.

I think if you fix the lock icon issue we can approve.

**Comment by leandroberetta** [comment]
> I think all of the prior comments are resolved! And we have decided to temporarily remove the tag hide/hover customization and contribute that feature to PFT itself. So that feature is expected to not be there.
> 
> There is a regression in showing lock icons on edge labels, this can be fixed by accepting the recommended changes in my comments.
> 
> Also, I'm seeing maybe an issue in trace overlay, but I don't think it has to do with this PR, it may be nothing, but it seems sometimes the overlay stops before the selected node. Anyway, not a blocker for this.
> 
> I think if you fix the lock icon issue we can approve.

Thanks for your review @jshaughn.

I vote for remove for now the hide/hover feature (there is a PR already for that in PF repo: https://github.com/patternfly/react-topology/pull/123, this issue is mentioned in the epic: https://github.com/kiali/openshift-servicemesh-plugin/issues/99). 

I will address the feedback.

**Pull request merged**
